metagraph/src/graph/representation/succinct/boss_chunk_construct.cpp

```diff
@@ -958,13 +953,16 @@ class BOSSChunkConstructor : public IBOSSChunkConstructor {
                          size_t disk_cap_bytes)
         : swap_dir_(swap_dir),
           kmer_collector_(k + 1,
-                          both_strands_mode,
+                          both_strands
+                            ? (filter_suffix.empty()
```

Comment:
maybe put a comment here to explain why the choice of `KmerCollector::Mode` depends on `filter_suffix`